### PR TITLE
feature/les 416 Edit video lesson title

### PR DESCRIPTION
**Pull request by OxygenO22**

### CODE REVIEW

File: src/components/LessonCard.tsx [hunk 1]
```diff
@@ -33,9 +34,13 @@ const Title: React.FC<TitleProps> = (props) => {
   );
 };
 
-const MenuKebab = () => {
+const MenuKebab = (CardId: any) => {
```
Comment (Kurilo-Pavel) [inline src/components/LessonCard.tsx:37]: type "any" is bad practice

File: src/components/editVideoLesson/EditVideoLessonForm.scss [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  width: 100%;
+  margin: $marginBottomMain;
+}
+
+.evlth__inner {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+}
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.scss:70]: add empty line

File: src/components/editVideoLesson/EditVideoLessonForm.scss [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  color: $warningColor;
+  font-weight: normal;
+  font-size: 12px;
+  margin-top: 5px;
+}
+
+@media (max-width: $max-width-3) {
+  .evlth__inner {
+    flex-direction: column;
+  }
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.scss:122]: add empty line

File: src/components/editVideoLesson/EditVideoLessonForm.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+import frame86 from "../icons/Frame86.png";
+import frame87 from "../icons/Frame87.png";
+import frame88 from "../icons/Frame88.png";
+import request from "../../services/request";
+import { useIntl } from "react-intl";
+
+
+const RegExpName = /^[а-яА-ЯёЁa-zA-Z0-9( )!$%&'""*+-/=?^_`{|}~.,@<>:[\]]+$/i;
+const DescriptionName =
+  /^[а-яА-ЯёЁa-zA-Z0-9( )!$%&'""*+-/=?^_`{|}~.,@<>:[\]#]+$/i;
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.tsx:22]: add in validationRules.ts

File: src/components/editVideoLesson/EditVideoLessonForm.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+const maxNameLength = 64;
+const maxDescriptionLength = 600;
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.tsx:24]: add in constants.ts

File: src/components/editVideoLesson/EditVideoLessonForm.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+    { id: 1, src: `${frame88}` },
+    { id: 2, src: `${frame87}` },
+    { id: 3, src: `${frame86}` },
+    { id: 4, src: `${frame85}` },
+  ];
+  const [lesson, setLesson] = useState<ILessonBack | null>(null);
+  const params = useParams();
+  const navigate = useNavigate();
+  const getLessonUrl =
+    "https://lessoner-project-2w3h.onrender.com/lessons/" + params.id;
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.tsx:41]: use constant in constants.ts

File: src/components/editVideoLesson/EditVideoLessonForm.tsx [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+  useEffect(() => {
+    return () => {
+      fetch(getLessonUrl)
+        .then((response) => response.json())
+        .then((lesson) => setLesson(lesson))
+        .catch((error) => console.log(error));
+    };
+  }, []);
+
+  const editL = (lessonFromEditForm: any) => {
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.tsx:54]: use "any" bad practic

File: src/components/editVideoLesson/EditVideoLessonForm.tsx [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+    return () => {
+      request(lessonEditUrl, "PUT", lessonFromEditForm);
+    };
+  };
+
+  const addInfoToLessonObject = (values: any) => {
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonForm.tsx:60]: use "any" is bad practice

File: src/components/editVideoLesson/EditVideoLessonTitle.scss [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+.evl__inner {
+  display: flex;
+  justify-content: center;
+  width: 100%;   
+  max-width: 656px;
+  min-height: 843px;
+  border-radius: 8px;
+  box-shadow: 1px 1px 20px rgba(46, 51, 99, 0.1);
+  background: #fff;
```
Comment (Kurilo-Pavel) [inline src/components/editVideoLesson/EditVideoLessonTitle.scss:20]: use constant

File: src/constants.scss [hunk 10]
```diff
@@ -46,6 +47,17 @@ $greySearchColor: #9A9AA3;
...
 }
 
+@mixin textFont {
+  & {
+    font-family: $font;
+    font-weight: 400;
+    font-size: 14px;
+    line-height: 150%;
+    font-style: normal;
+    color: #1B1B1F;
```
Comment (Kurilo-Pavel) [inline src/constants.scss:57]: use constant

File: src/components/editVideoLesson/EditVideoLessonForm.scss (hunk 2 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonForm.scss:70]: done

File: src/components/editVideoLesson/EditVideoLessonForm.scss (hunk 3 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonForm.scss:122]: done

File: src/components/editVideoLesson/EditVideoLessonForm.tsx (hunk 4 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonForm.tsx:22]: done

File: src/components/editVideoLesson/EditVideoLessonForm.tsx (hunk 5 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonForm.tsx:24]: done

File: src/components/editVideoLesson/EditVideoLessonTitle.scss (hunk 9 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonTitle.scss:20]: done

File: src/constants.scss (hunk 10 above)
Comment (OxygenO22) [reply src/constants.scss:57]: done

File: src/components/Button.tsx [hunk 11]
```diff
@@ -6,14 +6,18 @@ type ButtonProps = {
...
 
 const Button = ({
-  buttonType, buttonText, onClick, className
+  buttonType, buttonText, onClick, className, buttonIcon
 }: ButtonProps) => {
   return (
     <Fragment>
-      <button type={buttonType} className={className} onClick={onClick}>{buttonText}</button>
+      <button type={buttonType} className={className} onClick={onClick}>
+        {<>{buttonIcon}</>}
```
Comment (Terapool) [inline src/components/Button.tsx:18]: Do we really need React fragment <> here:

{<>{buttonIcon}</>} ?

Simply {buttonIcon} doesn`t work?

File: src/components/LessonCard.tsx (hunk 1 above)
Comment (Terapool) [reply src/components/LessonCard.tsx:37]: Yes, it is better to declare proper type for it

File: src/components/LessonCard.tsx [hunk 12]
```diff
@@ -96,7 +101,7 @@ const LessonCard: React.FC<LessonCardsProps> = (props) => {
         <div className="card__info">
           <div className="card__info-top">
             <Title title={props.title} />
-            <MenuKebab />
+            <MenuKebab idCard={props.id}/>
```
Comment (Terapool) [inline src/components/LessonCard.tsx:104]: And for pretty view you need a space before "/"
... idCard={props.id} /...

File: src/components/body/content/lessons/Lessons.tsx [hunk 13]
```diff
@@ -1,28 +1,55 @@
...
+  FC, useEffect, useState 
+} from "react";
 import { FormattedMessage } from "react-intl";
 import LessonCard from "../../../LessonCard";
-import React from "react";
-import items from "../../../../assets/lessonCard.json";
 
-const Lessons: React.FC = () => {
+const Lessons: FC = () => {
+  const [lessons, setLessons] = useState([]);
```
Comment (Terapool) [inline src/components/body/content/lessons/Lessons.tsx:9]: Import lesson type from lesson component and use it for lessons, e.g.

const [lessons, setLessons] = useState&lt;LessonType&gt;([]);

File: src/components/body/content/lessons/Lessons.tsx [hunk 14]
```diff
@@ -1,28 +1,55 @@
...
-            published={obj.published}
-            view={obj.view}
-            category={obj.category}
-            rating={obj.rating}
-            totalVotes={obj.totalVotes}
-          />))
-        }
+        {lessons.map(
+          (obj: {
+            id: number;
```
Comment (Terapool) [inline src/components/body/content/lessons/Lessons.tsx:27]: You do not need to write here lesson type, you need to import this type. It is already exists

**Review by Terapool — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx (hunk 11 above)
Comment (OxygenO22) [reply src/components/Button.tsx:18]: :) It's interesting fact, earlier I tried to use {buttonIcon}, and it doesn't work, but now it work. Done

File: src/components/LessonCard.tsx (hunk 12 above)
Comment (OxygenO22) [reply src/components/LessonCard.tsx:104]: Done

File: src/components/LessonCard.tsx (hunk 1 above)
Comment (OxygenO22) [reply src/components/LessonCard.tsx:37]: done

File: src/components/editVideoLesson/EditVideoLessonForm.tsx (hunk 8 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonForm.tsx:55]: done

File: src/components/body/Body.scss [hunk 15]
```diff
@@ -0,0 +1,11 @@
+@import "../../constants.scss";
+
+.body {
+  box-sizing: border-box;
+    display: flex;
```
[the remark this answers is not in the document]
Comment (OxygenO22) [reply src/components/body/Body.scss:5]: done

File: src/components/editVideoLesson/EditVideoLessonForm.tsx (hunk 6 above)
Comment (OxygenO22) [reply src/components/editVideoLesson/EditVideoLessonForm.tsx:36]: done

File: src/components/body/content/lessons/Lessons.tsx (hunk 13 above)
Comment (OxygenO22) [reply src/components/body/content/lessons/Lessons.tsx:9]: done

File: src/components/body/content/lessons/Lessons.tsx (hunk 14 above)
Comment (OxygenO22) [reply src/components/body/content/lessons/Lessons.tsx:27]: done

**Review by valer-sky — APPROVED** [review]

**Review by Terapool — APPROVED** [review]

**Review by Kurilo-Pavel — APPROVED** [review]

**Pull request merged**